client/engine/engine.go

```diff
@@ -261,3 +272,26 @@ func (e *Engine) constructObjectiveFromMessage(message protocols.Message) (proto
 	}
 
 }
+
+type messageSummary struct {
+	to           string
+	objectiveId  protocols.ObjectiveId
+	signedStates []signedStateSummary
+}
+type signedStateSummary struct {
+	turnNum   uint64
+	channelId string
+}
+
+// summarizeMessage returns a basic summary of a message suitable for logging
+func summarizeMessage(message protocols.Message) messageSummary {
+	summary := messageSummary{to: message.To.String(), objectiveId: message.ObjectiveId, signedStates: []signedStateSummary{}}
+	for _, signedState := range message.SignedStates {
+		channelId, err := signedState.State().ChannelId()
+		if err != nil {
+			panic(err)
+		}
+		summary.signedStates = append(summary.signedStates, signedStateSummary{turnNum: signedState.State().TurnNum, channelId: channelId.String()})
+	}
+	return summary
+}
```

Comment:
This is great. As a refactor (separate PR is fine), maybe it should live in `protocols/messages.go`?